presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsBucketFunction.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tpcds;
+
+import com.facebook.presto.spi.BucketFunction;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.google.common.primitives.Ints;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public class TpcdsBucketFunction
+        implements BucketFunction
+{
+    private final int bucketCount;
+    private final long rowsPerBucket;
+
+    public TpcdsBucketFunction(int bucketCount, long rowsPerBucket)
+    {
+        this.bucketCount = bucketCount;
+        this.rowsPerBucket = rowsPerBucket;
+    }
+
+    @Override
+    public int getBucket(Page page, int position)
+    {
+        Block block = page.getBlock(0);
+        if (block.isNull(position)) {
+            return 0;
+        }
+
+        long rowNumber = BIGINT.getLong(block, position);
+        int bucket = Ints.checkedCast(rowNumber / rowsPerBucket);
```

Comment:
wrap with`try-catch` and throw proper `PrestoException` instead of generic `ArithmeticException`.